python/cudf/cudf/core/column/string.py

```diff
@@ -1680,17 +1720,10 @@ def find(
             end,
         )
         res = type(self).from_pylibcudf(plc_result)
-        if not is_pandas_nullable_extension_dtype(self.dtype) and (
-            end is None or end == -1
-        ):
-            res = res._with_type_metadata(
-                get_dtype_of_same_kind(self.dtype, res.dtype)
-            )
-        else:
-            res = res.astype(np.dtype("int64"))  # type: ignore[assignment]
-            res = res._with_type_metadata(
-                get_dtype_of_same_kind(self.dtype, np.dtype("int64"))
-            )
+        if cudf.get_option("mode.pandas_compatible"):
```

Comment:
Found a bug we can turn into a test
```python
In [1]: %load_ext cudf.pandas

In [2]: import pandas as pd

In [3]: pd.Series(["foo", "bar", None], dtype=object).str.find("o").dtype
Out[3]: dtype('int64')

In [4]: quit
(rapids) coder ➜ ~/cudf $ ipython
Python 3.13.9 | packaged by conda-forge | (main, Oct 22 2025, 23:33:35) [GCC 14.3.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 9.7.0 -- An enhanced Interactive Python. Type '?' for help.
Tip: You can find how to type a LaTeX symbol by back-completing it, eg `\θ<tab>` will expand to `\theta`.

In [1]: import pandas as pd

In [2]: pd.Series(["foo", "bar", None], dtype=object).str.find("o").dtype
Out[2]: dtype('float64')
```

Reply:
```
Returns:

    Series or Index of int.
```
Nevermind, might be a pandas bug. See https://pandas.pydata.org/docs/reference/api/pandas.Series.str.find.html

Reply:
Ill file a pandas bug for this